src/AutomatedCar/Helpers/CollisionDetection.cs

```diff
@@ -165,7 +166,29 @@ public static bool PointInTriangle(Point point, Tuple<Point, Point, Point> trian
             return dotProduct1 >= 0 && dotProduct2 >= 0 && dotProduct3 >= 0;
         }
 
-        private static Point RotatePoint(Point point, double angle)
+        /// <summary>
+        /// Returns whether point is in a triangle with Barycentric method.
+        /// </summary>
+        /// <param name="p">Point</param>
+        /// <param name="p0">Triangle Point 1.</param>
+        /// <param name="p1">Triangle Point 2.</param>
+        /// <param name="p2">Triangle Point 3.</param>
+        /// <returns>Point is in triangle or not.</returns>
+        public static bool PointInTriangle(Point p, Point p0, Point p1, Point p2)
```

Comment:
Ne legyen kettő PointInTriangle metódus. Ha ez fixálja a másik hibáit, akkor:
1. Töröld ki másikat
2. A háromszög egy Point Tuple legyen, részben backward compatibility miatt, részben azért, mert a Test Suite arra a függvényre van írva.
3. Summaryban kommentáld edge caseket.

Reply:
És légyszi pullolj developról kövi commit előtt, Collision Detection 1 verzióval előrébb van, mint a tied.

Reply:
Javítom, 10 perc, viszont ez azelőtt íródott, mielőtt az a pull request el lett fogadva.